app/src/main/java/com/conupods/OutdoorMaps/BuildingDataMap.java

```diff
@@ -0,0 +1,83 @@
+package com.conupods.OutdoorMaps;
+
+import android.content.res.AssetManager;
+import android.util.Log;
+
+import com.conupods.App;
+import com.google.android.gms.maps.model.LatLng;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+
+/**
+ * A singleton class that holds hardcoded building data of concordia buildings.
+ * The class parses the data (only once) that is saved in the file ../assets/buildings.json
+ * who was imported from the concordia API manually.
+ */
+public class BuildingDataMap {
+    private static BuildingDataMap mInstance;
+    private HashMap<LatLng, Building> mData;
+    private static final String TAG = "BUILDING_DATA_MAP";
+
+    private BuildingDataMap() throws BuildingException {
+        // todo possibly change the index to be the marker object and not latlng
+        mData = new HashMap<>();
+        parseBuildingData();
+    }
+
+    public static synchronized BuildingDataMap getInstance() {
+        if (mInstance == null) {
```

Comment:
what if you use eager initialization? the synchronized will add some overhead, and we know for a fact that we will need this stuff at the startup of the app. so while the loading screen is running, this code can be ran.